Client/Frontend/AuthenticationManager/AuthenticationSettingsViewController.swift

```diff
@@ -115,28 +115,32 @@ extension RequirePasscodeSetting: PasscodeEntryDelegate {
     }
 }
 
-class TouchIDSetting: Setting {
-    private let authInfo: AuthenticationKeychainInfo?
+class AuthenticationSetting: Setting {
+    private var authInfo: AuthenticationKeychainInfo?
 
     private weak var navigationController: UINavigationController?
     private weak var switchControl: UISwitch?
 
-    private var touchIDSuccess: (() -> Void)? = nil
-    private var touchIDFallback: (() -> Void)? = nil
+    private var touchIDSuccess: (AuthenticationSetting -> Void)? = nil
+    private var touchIDFallback: (AuthenticationSetting -> Void)? = nil
+    
+    var authenticationDelegate: AuthenticationSettingDelegate!
 
     init(
         title: NSAttributedString?,
         navigationController: UINavigationController? = nil,
         delegate: SettingsDelegate? = nil,
+        authenticationDelegate: AuthenticationSettingDelegate! = nil,
```

Comment:
This worries be since explictly unwrapped variables should never be nil. If there is a possibility of it being nil, like in the default case, I would change it to a regular optional and either use optional changing (`authenticationDelegate?.blah()`) or guard statements to invoke methods.